tarofreighter/chain_porter.go

```diff
@@ -600,6 +600,16 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 		// to complete the send w/o merging inputs.
 		assetInput := elgigibleCommitments[0]
 
+		// If the key found for the input UTXO is not from the Taro
+		// keyfamily, something has gone wrong with the DB.
+		if assetInput.InternalKey.Family != tarogarden.TaroKeyFamily {
```

Comment:
Did we run into this somewhere? Or what's the reason for this specific check? Mostly just curious.

Reply:
I think was added during the debugging phase as a sort of sanity check to help track down the bug.